Libraries/LibWeb/CSS/Parser/PropertyParsing.cpp

```diff
@@ -1653,6 +1658,91 @@ RefPtr<CSSStyleValue const> Parser::parse_border_value(PropertyID property_id, T
         { border_width.release_nonnull(), border_style.release_nonnull(), border_color.release_nonnull() });
 }
 
+// https://drafts.csswg.org/css-backgrounds/#border-image-slice
+RefPtr<CSSStyleValue const> Parser::parse_border_image_slice_value(TokenStream<ComponentValue>& tokens)
+{
+    // [<number [0,∞]> | <percentage [0,∞]>]{1,4} && fill?
+    auto transaction = tokens.begin_transaction();
+    auto fill = false;
+    RefPtr<CSSStyleValue const> top;
+    RefPtr<CSSStyleValue const> left;
+    RefPtr<CSSStyleValue const> bottom;
+    RefPtr<CSSStyleValue const> right;
+
+    auto parse_fill = [&](TokenStream<ComponentValue>& fill_tokens) -> Optional<bool> {
+        if (auto keyword = parse_keyword_value(fill_tokens)) {
+            if (fill || keyword->to_keyword() != Keyword::Fill)
+                return {};
+            return true;
+        }
+        return false;
+    };
+
+    auto maybe_fill_value = parse_fill(tokens);
+    if (!maybe_fill_value.has_value())
+        return nullptr;
+    if (*maybe_fill_value)
+        fill = true;
+
+    Vector<ValueComparingNonnullRefPtr<CSSStyleValue const>> number_percentages;
+    while (number_percentages.size() <= 4 && tokens.has_next_token()) {
+        auto number_percentage = parse_number_percentage_value(tokens);
+        if (!number_percentage)
+            break;
+
+        if (number_percentage->is_number() && !property_accepts_number(PropertyID::BorderImageSlice, number_percentage->as_number().number()))
+            return nullptr;
+        if (number_percentage->is_percentage() && !property_accepts_percentage(PropertyID::BorderImageSlice, number_percentage->as_percentage().percentage()))
+            return nullptr;
+        number_percentages.append(number_percentage.release_nonnull());
+    }
+
+    switch (number_percentages.size()) {
+    case 1:
+        top = number_percentages[0];
+        left = number_percentages[0];
+        bottom = number_percentages[0];
+        right = number_percentages[0];
+        break;
+    case 2:
+        top = number_percentages[0];
+        bottom = number_percentages[0];
+        left = number_percentages[1];
+        right = number_percentages[1];
+        break;
+    case 3:
+        top = number_percentages[0];
+        left = number_percentages[1];
+        right = number_percentages[1];
+        bottom = number_percentages[2];
+        break;
+    case 4:
+        top = number_percentages[0];
+        left = number_percentages[1];
+        bottom = number_percentages[2];
+        right = number_percentages[3];
```

Comment:
> If there are four values they apply to the top, right, bottom, and left, respectively.

Apart from correcting this, it's probably worth putting the variables and assignments in that order everywhere.

Reply:
I've fixed this by swapping `left` and `right` values everywhere. I've kept the order such that like values are grouped together, since IMO it's easier to see what is going on that way. 

If you prefer I can always assign in trbl order instead.